tutorials/2023-11-06-offchain-sdk-ts-generation-with-cpm/index.md

```diff
@@ -0,0 +1,110 @@
+---
+title: 'How to use CPM to generate an off-chain TypeScript SDK'
+description: "The tutorial aims to provide a comprehensive guide for developers interested in creating TypeScript SDKs for Neo smart contracts, making it easier for them to leverage CPM and successfully work with off-chain SDKs."
+author: coz.io
+tags: ["TS", "CPM", "SDK"]
+skill: BEGINNER
+sidebar: true
+---
+
+[CPM](https://github.com/CityOfZion/cpm) is a tool that can help you download smart contracts and generate on-chain and off-chain SDKs for multiple languages without the need to write them yourself and needing none or minimal changes.
+
+## 1. Initializing workspace
+We recommend using [vite](https://vitejs.dev/guide/) to create your project from a template. You just need to run `npm create vite@latest` and select the prompts to generate your project. After creating your project, it's time to start using CPM to automate the off-chain SDK generation.
+
+## 2. Using CPM 
+It's pretty easy to use CPM to add an off-chain SDK to your TypeScript project. After installing CPM, you can run it with a `cpm.yaml` or you could directly use the command prompt. In this tutorial, we will be using the easiest one to maintain.
```

Comment:
```suggestion
It's pretty easy to use CPM to add an off-chain SDK to your TypeScript project. After installing CPM you can generate a single SDK directly from the command prompt or setup a configuration file that will be processed and allows for generating multiple SDKs at the same time. In this tutorial, we will be using the easiest one to maintain.
```

The reason for my suggestion is that in the original text 
> you can run it with a `cpm.yaml`

`cpm.yaml` is mentioned for the first time in the article without having explained what it is. Not everybody might know that this is the configuration file.